tests/browser_e2e_tests/create_and_query_database.rs

```diff
@@ -0,0 +1,222 @@
+use crate::utils::browser::BrowserHelpers;
+use playwright::api::{page, Page};
+use std::error::Error;
+use std::fs;
+use std::path::Path;
+
+pub async fn test_create_and_query_database_flow(
+    page: &Page,
+    username: &str,
+) -> Result<(), Box<dyn Error>> {
+    // Step 1: Verify we're on the user dashboard and take initial screenshot
+    // After profile test, the display name has changed from username to "Entity 0"
+    let expected_title = "Entity 0 - ayb";
+    assert_eq!(page.title().await?, expected_title);
+
+    BrowserHelpers::screenshot_compare(&page, "dashboard_before_database_creation", &[]).await?;
+
+    // Step 2: Click on "Create database" button
+    page.click_builder("button:has-text('Create database')")
+        .timeout(3000.0)
+        .click()
+        .await?;
+
+    // Screenshot of the create database form
+    BrowserHelpers::screenshot_compare(&page, "create_database_form", &[]).await?;
+
+    // Step 3: Fill in database name (using similar name to e2e tests)
+    page.fill_builder("input[name='database_slug']", "test.sqlite")
+        .timeout(1000.0)
+        .fill()
+        .await?;
+
+    // Screenshot of filled form
+    BrowserHelpers::screenshot_compare(&page, "database_form_filled", &[]).await?;
+
+    // Step 4: Submit the create database form
+    page.click_builder("button[type='submit']:has-text('Create database')")
+        .timeout(5000.0)
+        .click()
+        .await?;
+
+    // Screenshot after database creation
+    BrowserHelpers::screenshot_compare(&page, "database_created", &[]).await?;
+
+    // Step 5: Ensure we're on the database page
+    let database_page_title = format!("Explore {}/test.sqlite - ayb", username);
+
+    // Verify we're on the database page
+    assert_eq!(page.title().await?, database_page_title);
+
+    // Screenshot of database page
```

Comment:
Given the screenshot on line 43, we can delete this one (and the reference screenshot in test/screenshots). We'll have to renumber the rest of the screenshots.